src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,227 @@
+package duke.command;
+
+import duke.exceptions.DukeIllegalDescriptionException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import java.util.Date;
+
+/**
+ * This is a class to make sense of the user commands and allow Duke to take different actions
+ * depending on the commands received.
+ */
+class Parser {
+
+    private String[] action;
+
+    /**
+     * Constructor of the Parser class.
+     *
+     * @param action A string array contains split commands.
+     */
+    Parser(String[] action) {
+        this.action = action;
+    }
+
+    /**
+     * To make amendments according to the first index of array which is the command type.
+     *
+     * @throws DukeIllegalDescriptionException To deal with the illegal and invalid inputs from users.
+     */
+    void parse() throws DukeIllegalDescriptionException {
+        switch (Command.valueOf(action[0])) {
+        case bye:
+            commandBye();
+            break;
+        case list:
+            commandList();
+            break;
+        case done:
+            commandDone();
+            break;
+        case todo:
+            commandTodo();
+            break;
+        case deadline:
+            commandDeadline();
+            break;
+        case event:
+            commandEvent();
+            break;
+        case delete:
+            commandDelete();
+            break;
+        case find:
+            commandFind();
+            break;
+        }
+
+    }
+
+    /**
+     * Action taken when the command is bye.
+     */
+    private void commandBye() {
+        System.out.println("Bye. Hope to see you again soon!");
+        Ui.setFlag();
+    }
+
+    /**
+     * Action taken when the command is list.
+     */
+    private void commandList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < TaskList.getList().size(); ++i) {
+            System.out.println(i + 1 + "." + TaskList.getList().get(i));
+        }
+    }
+
+    /**
+     * Action taken when the command is done.
+     */
+    private void commandDone() {
+        Storage storage = new Storage();
+        int num = Integer.parseInt(action[1]);
+        Task newTask = TaskList.getList().get(num - 1);
```

Comment:
Maybe you can also try catching index out of bounds exception here: 
Suppose we currently have 3 items in the task list. If the user made a fat finger error and typed "done 4" this would throw the out of bounds error and unintentionally terminate the program. So maybe it would be helpful to catch the error and then print something like "Ensure task number is between 1 and " + TaskList.getList().size() to remind the user what the range of valid task numbers are :)

Reply:
Thank you for the suggestions! I will add them on.